### PR TITLE
feat(core): DebugNode.triggerEventHandler should accept eventObj as an optional parameter

### LINKED ISSUE #44724: make `DebugNode.triggerEventHandler` `eventObj` a proper optional parameter
### Which @angular/* package(s) are relevant/releated to the feature request?

core

### Description

This is to resurrect a change requested here years ago: #10829
The change is minor, but impacts everyone writing tests:
```ts
triggerEventHandler(eventName: string, eventObj: any): void;
```

The current impementation is too terse and forces everyone either to put enything 'sane' or just give up and use whatever quicker, like '{}', including Angular own documentation and test code (this is only from repo code):
```ts
input.triggerEventHandler('keyup', { });
heroDe.triggerEventHandler('click', null);
fixture.debugElement.children[1].triggerEventHandler('myevent', <Event>{});
 ```
 This prevents most adding a quick test for a common code like, again, Angular docs:
 ```html
 <button (click)="save()">Save</button>
 <button (click)="cancel()">Cancel</button>
  ```
 to be tested without switching to additional mode of coding stuff (and adding whatever person writing a code thinks is OK for 'eventObj'). Docs contains special section on 'why' this is required:
https://angular.io/guide/testing-components-scenarios#trigger-event-handler
 
 Thanks!

### Proposed solution

```ts
// as non-transpiled TS
triggerEventHandler(eventName: string, eventObj: any = null): void {
// as declaration type with `eventObj` desribed as `[eventObj=null]`
triggerEventHandler(eventName: string, eventObj?: any): void;
```

If accepted, the change would require review and changes accross different places and documentation, including change to 'Whys' it was added:
https://angular.io/guide/testing-components-scenarios#trigger-event-handler

### Alternatives considered

None

### PR DESCRIPTION
**Pull request by JiaLiPassion**
Close https://github.com/angular/angular/issues/44724

`DebugNode.triggerEventHandler()` should accept the `eventObj` as an
optional parameter. So the user don't have to write code like

```
elem.triggerEventHandler('click', null);
```

### CODE REVIEW

File: aio/content/examples/testing/src/app/dashboard/dashboard-hero.component.spec.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
 
 // #docplaster
-import { DebugElement } from '@angular/core';
-import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
-import { By } from '@angular/platform-browser';
-import { first } from 'rxjs/operators';
+import {DebugElement} from '@angular/core';
```
Comment (crisbeto) [inline aio/content/examples/testing/src/app/dashboard/dashboard-hero.component.spec.ts:3]: Should these formatting changes be reverted? I think that AIO follows different formatting rules than the rest of the repo.

Comment (JiaLiPassion) [reply aio/content/examples/testing/src/app/dashboard/dashboard-hero.component.spec.ts]: @crisbeto , thank you for the review, I reverted the format.

**Review by crisbeto — APPROVED** [review]
LGTM

**Review by AndrewKushnir — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Review by atscott — APPROVED** [review]
Reviewed-for: public-api

**Review by dylhunn — APPROVED** [review]
reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:433383655:BASE:433383667:1646801819745:d1d24f8a).

**Comment by atscott** [comment]
This PR was merged into the repository by commit c7cab69af1ef3ff6bf81b8075801031d16914e6f.

**Comment by peterblazejewicz** [comment]
thx!

**Pull request closed without merging**